src/CloudNative.CloudEvents/CloudEvent.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Cloud Native Foundation. 
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+namespace CloudNative.CloudEvents
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Mime;
+
+    public class CloudEvent
+    {
+        readonly IDictionary<string, object> attributes;
+
+        /// <summary>
+        /// Create a new CloudEvent instance.
+        /// </summary>
+        /// <param name="type">'type' of the CloudEvent</param>
+        /// <param name="source">'source' of the CloudEvent</param>
+        /// <param name="id">'id' of the CloudEvent</param>
+        /// <param name="time">'time' of the CloudEvent</param>
+        /// <param name="extensions">Extensions to be added to this CloudEvents</param>
+        public CloudEvent(string type, Uri source, string id = null, DateTime? time = null,
+            params ICloudEventExtension[] extensions) : this(extensions)
+        {
+            Type = type;
+            Source = source;
+            Id = id ?? Guid.NewGuid().ToString();
+            Time = time ?? DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Create a new CloudEvent instance
+        /// </summary>
+        /// <param name="extensions">Extensions to be added to this CloudEvents</param>
+        internal CloudEvent(params ICloudEventExtension[] extensions)
+        {
+            attributes = new CloudEventAttributes(extensions);
+            SpecVersion = "0.1";
```

Comment:
should this, with things like `id` (instead of `eventId`), be 0.2 ?

Reply:
Yes. I'll fix that :)